lib/shellable.ts

```diff
@@ -186,6 +186,55 @@ export interface AssumeRole {
    * underscores or  any  of  the  following characters: =,.@:/-
    */
   externalId?: string;
+
+  /**
+   * When a profie name is configured, an assumed role configuration will be created
+   * in the shared aws configuration file (~/.aws/config). This is in contrary of simply invoking
+   * an `sts assume-role` command that creates a session with a fixed expiry date.
+   *
+   * Using a profile will delegate credential refreshing to the SDK/CLI.
+   * This is needed to support long running sessions that needs sessions that are longer than
+   * the session duration that can be configured with a `sts assume-role`.
+   *
+   * When a profile name is specified, you must also configure one of `sourceProfile` or `credentialSource`.
+   *
+   * @see https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html
+   */
+  profileName?: string,
+
+  /**
+   * A profile name to be used as the credential with which to assume the role.
+   * Must exist in either the credential or the config file.
+   * Either this or `credentialSource` is required when using `profileName`
+   *
+   * @see https://docs.aws.amazon.com/credref/latest/refdocs/setting-global-source_profile.html
```

Comment:
What's the `@default` behavior?

I'd personally still like it if we could just write to the `[default]` profile!

I know that you're triggering the profile-vs-environment variables behavior on the presence of this property.

But how about we add `needsLongSession?: boolean`, trigger the behavior off of that and use `[default]` as the default value for profile if not given?